### DISCUSSION TITLE
Certificate for courses.openmined.com

### DISCUSSION THREAD

**Discussion by cescapa** [OpenMined/courses#207] [Feedback] [answered]
When I click in the icon lock for courses.openmined.org, a certificate appears for web site malvika.info. Please see image below. This web site belongs to an undergraduate student in Duke University, which is located in North Carolina US. How and why is this certificate associated with OpenMined? 

![certificate openmined](https://user-images.githubusercontent.com/8942908/104854514-f192f200-58bb-11eb-87ba-a646a49b49b6.jpg)

**Comment by iamtrask** [comment]
Fascinating - will get to the bottom of this ASAP

**Comment by iamtrask** [comment]
Fwiw - so far I am unable to reproduce this error in my own browser. I also did a search through out codebase and cannot find anything relating to that domain.

**Comment by majauhar** [comment]
Here is what I see on my computer
![Screenshot 2021-01-18 at 8 05 10 PM](https://user-images.githubusercontent.com/43479350/104928464-b500ec80-59c8-11eb-98b1-c0f50c53a97d.png)

**Comment by jaintj95** [comment]


**Reply by majauhar** [reply]
Mine is still the malvika one!

**Comment by cereallarceny** [comment, marked as answer]
I looked into this and it appears to actually be the strange, but intended, functionality here. The Courses website is hosted on Firebase which will provide a shared SSL certificated to all websites hosted there. Technically speaking though, they don't provision a unique SSL certificate for each website on their platform, but instead share the certificates with other people using Firebase. They do this using "Subject Alternative Names", which is fancy SSL talk for "all the other domains using this SSL cert". It appears that the Firebase infrastructure team either a) got lazy and named the certificated as the first domain name used for that cert, or b) made a mistake. But either way, this is the expected functionality since we don't use our SSL cert for the courses site that we use on our main website (hosted on AWS).

You can read more here: https://firebase.google.com/docs/hosting/custom-domain#ssl-provisioning

**Reply by cereallarceny** [reply]
Top of the list:
![Screen Shot 2021-01-18 at 6 02 57 PM](https://user-images.githubusercontent.com/1297930/104950689-0d2ff280-59b9-11eb-916b-290071b111c5.png)

Our record a bit further down:
![Screen Shot 2021-01-18 at 6 03 08 PM](https://user-images.githubusercontent.com/1297930/104950694-0ef9b600-59b9-11eb-9710-6efe41536ea6.png)

**Reply by cescapa** [reply]
this is great cybersleuth work, thank you Patrick for the explanation
at least we now know that this isn't some funny man-in-the-middle highjack which was my initial concern
I am surprised that the Google/Firebase team would just pick the first domain name; I bet you that both options (a) and (b) apply here

**Reply by cereallarceny** [reply]
Happy to help! Quite honestly when I saw it, my first thought was "oh no, I'm in big trouble..."

**Reply by cereallarceny** [reply]
@cescapa @iamtrask @jaintj95 @majauhar I spoke to the Firebase team and they were willing to stamp out a certificate specifically for our domain names - how nice! Just thought I'd update you all if you were curious, the website should now show the cert is registered to "courses.openmined.org".

**Reply by cescapa** [reply]
I confirm! Well done Patrick!!

**Reply by majauhar** [reply]
Cool! I see it got issued on 02nd Feb. Great work @cereallarceny 😃
